node/block.go

```diff
@@ -27,7 +27,7 @@ func (node *Node) AddNewBlock(ctx context.Context, b *types.Block) (err error) {
 	}
 
 	log.Debugf("syncing new block: %s", b.Cid().String())
-	if err := node.Syncer.HandleNewTipset(ctx, types.NewTipSetKey(blkCid)); err != nil {
+	if err := node.Syncer.HandleNewTipset(ctx, types.NewTipSetKey(blkCid), node.Host().ID()); err != nil {
```

Comment:
Thinking out loud: It would be nice if the fetcher logic could be skipped for this case, else we'll have to check if this peerID is our own before fetching from it -- that can all happen later though.